RELEASES.md

```diff
@@ -0,0 +1,149 @@
+# Releases
+
+Tendermint uses [semantic versioning](https://semver.org/) with each release following
+a `vX.Y.Z` format. The `master` branch is used for active development and thus it's
+advisable not to build against it.
+
+Releases are specified using tags and are built from long-lived "backport" branches.
```

Comment:
Let's clarify here that the latest changes and features are always merged into `master`.